pallets/claim/src/lib.rs

```diff
@@ -0,0 +1,467 @@
+        /// Genesis balance for this pallet's account
+        pub genesis_balance: BalanceOf<T>,
+    }
+
+    #[pallet::genesis_build]
+    impl<T: Config> BuildGenesisConfig for GenesisConfig<T> {
+        fn build(&self) {
+            use frame_support::assert_ok;
+
+            // Create Claim account
+            let account_id = <Pallet<T>>::account_id();
+
+            // Fill account with genesis balance
+            let min = T::Currency::minimum_balance();
+            let _ = T::Currency::mint_into(&account_id, min.saturating_add(self.genesis_balance));
+
+            TotalClaimable::<T>::put(BalanceOf::<T>::zero());
+
+            // Add beneficiaries
+            if !self.beneficiaries.is_empty() {
+                assert_ok!(<Pallet<T>>::do_add_beneficiaries(
+                    self.beneficiaries.clone().into_iter().collect()
+                ));
+
+                // Initialize other storage variables
+                AirdropActive::<T>::put(true);
+                AirdropId::<T>::put(0);
+            }
+        }
+    }
+
+    #[pallet::event]
+    #[pallet::generate_deposit(pub(super) fn deposit_event)]
+    pub enum Event<T: Config> {
+        /// Beginning of a new airdrop campaign
+        AirdropStarted {
+            /// The id of the airdrop that has just started
+            airdrop_id: u64,
+        },
+        /// Some amount has been claimed by the beneficiary
+        Claimed {
+            /// Who claimed the tokens
+            beneficiary: T::AccountId,
+            /// How many tokens were claimed
+            amount: BalanceOf<T>,
+            /// The payment id
+            payment_id: <T::Paymaster as Pay>::Id,
+        },
+        /// Ending of the airdrop campaign
+        AirdropEnded {
+            /// The id of the airdrop that has just ended
+            airdrop_id: u64,
+        },
+    }
+
+    /// Error for the claim pallet.
+    #[pallet::error]
+    pub enum Error<T> {
+        /// Attemp to start a new airdrop while there is one already in progress
+        AlreadyStarted,
+        /// The pot has not enough funds available to cover for all the airdropped amounts
+        NotEnoughFunds,
+        /// Account requested a claim but it is not present among the Beneficiaries
+        NotEligible,
+        /// Added a beneficiary without balance to claim
+        NothingToClaim,
+        /// Attempt to modify the balance of an already added beneficiary
+        AlreadyPresent,
+        /// There was some issue with the mechanism of payment.
+        PayoutError,
+        /// Attempt to perform an action implying an open airdrop, while it has already ended
+        AlreadyEnded,
+    }
+
+    impl<T: Config> Pallet<T> {
+        /// The account ID of the claim pot.
+        ///
+        /// This actually does computation. If you need to keep using it, then make sure you cache the
+        /// value and only call this once.
+        pub fn account_id() -> T::AccountId {
```

Comment:
According to the comment save it in a storage value to not recompute it every time ... or remove the comment.

Reply:
define
```rust
    #[pallet::storage]
    pub type AccountId<T: Config> = StorageValue<_, T::AccountId>;
```
now you can use
```rust
    AccountId::<T>::mutate(|v| {
        let id = v.unwrap_or_else(|| T::PalletId::get().into_account_truncating());
        if v.is_none() {
            *v = Some(id.clone());
        }
        id
    })
```
I don't know if there is a simpler way to do it... If I'm not remember wrong I saw it implemented in a better way in pallet-timestamp or pallet-autority

Reply:
I took a look to `pallet_authorship` and it implements like follow:

```rust
		// Check the memorized storage value.
		if let Some(id) = AccountId::<T>::get() {
			return id
		}

		let id = T::PalletId::get().into_account_truncating();
        AccountId::<T>::put(&id);
        id
```